### PR TITLE
section commit B for July 7th, '20

### PR DESCRIPTION
**Pull request by croqueGrec09**
OrgProperty refactored; migration of CustomProperty and merge of OrgPrivateProperty into OrgProperty

### CODE REVIEW

File: app/grails-app/domain/com/k_int/properties/PropertyDefinition.groovy [hunk 1]
```diff
@@ -308,7 +308,7 @@ class PropertyDefinition extends AbstractI10nOverride implements Serializable, C
         def ownerClassName = classString.substring(classString.lastIndexOf(".") + 1)
         boolean isPublic
 
-        if(owner.class.name in [Subscription.class.name, License.class.name]) {
+        if(!owner.hasProperty("privateProperties")) {
```
Comment (Moetez16) [inline app/grails-app/domain/com/k_int/properties/PropertyDefinition.groovy:311]: privateProperties?

Comment (croqueGrec09) [reply app/grails-app/domain/com/k_int/properties/PropertyDefinition.groovy:311]: Is a bridge for that functionality works for both unmigrated and migrated properties. In one of the next pull requests, this check gets obsolete.

File: app/grails-app/controllers/de/laser/YodaController.groovy [hunk 2]
```diff
@@ -1698,7 +1699,7 @@ class YodaController {
         if (params.cmd == 'doIt') {
             println opp.collect{ it.id }
             if (opp.size() > 0) {
-                OrgPrivateProperty.executeUpdate('DELETE FROM OrgPrivateProperty opp WHERE opp.id in :idList',
+                OrgProperty.executeUpdate('DELETE FROM OrgProperty opp WHERE opp.id in :idList',
```
Comment (Moetez16) [inline app/grails-app/controllers/de/laser/YodaController.groovy:1702]: Where isPublic = false

File: app/grails-app/controllers/de/laser/YodaController.groovy [hunk 3]
```diff
@@ -764,36 +764,36 @@ class YodaController {
     def migrateNatStatSettings() {
         Map<String, Object> result = [:]
 
-        List<OrgCustomProperty> ocpList = OrgCustomProperty.executeQuery(
-                'select ocp from OrgCustomProperty ocp join ocp.type pd where pd.descr = :orgConf', [
+        List<OrgProperty> opList = OrgProperty.executeQuery(
+                'select op from OrgProperty op join op.type pd where pd.descr = :orgConf', [
```
Comment (Moetez16) [inline app/grails-app/controllers/de/laser/YodaController.groovy:768]: Where isPublic = true

File: app/grails-app/domain/com/k_int/properties/PropertyDefinition.groovy (hunk 1 above)
Comment (Moetez16) [reply app/grails-app/domain/com/k_int/properties/PropertyDefinition.groovy:311]: okay

File: app/grails-app/controllers/com/k_int/kbplus/OrganisationController.groovy [hunk 4]
```diff
@@ -702,8 +702,8 @@ class OrganisationController extends AbstractDebugController {
 
         List<PropertyDefinition> mandatories = PropertyDefinition.getAllByDescrAndMandatoryAndTenant(PropertyDefinition.ORG_PROP, true, result.institution)
 
-        mandatories.each { pd ->
-            if (!OrgPrivateProperty.findWhere(owner: result.orgInstance, type: pd)) {
+        mandatories.each { PropertyDefinition pd ->
```
Comment (Moetez16) [inline app/grails-app/controllers/com/k_int/kbplus/OrganisationController.groovy:705]: Where isPublic = false

File: app/grails-app/controllers/de/laser/YodaController.groovy (hunk 2 above)
Comment (croqueGrec09) [reply app/grails-app/controllers/de/laser/YodaController.groovy:1702]: Should be filtered by idList

File: app/grails-app/controllers/de/laser/YodaController.groovy (hunk 3 above)
Comment (croqueGrec09) [reply app/grails-app/controllers/de/laser/YodaController.groovy]: fixed in b44cda1

File: app/grails-app/controllers/de/laser/YodaController.groovy [hunk 5]
```diff
@@ -764,36 +764,36 @@ class YodaController {
...
                 else {
-                    oss.setValue(ocp)
+                    oss.setValue(op)
                 }
             }
         }
 
-        OrgCustomProperty.executeQuery(
-                'select ocp from OrgCustomProperty ocp join ocp.type pd where pd.descr = :orgConf '
+        OrgProperty.executeQuery(
```
[the remark this answers is not in the document]
Comment (croqueGrec09) [reply app/grails-app/controllers/de/laser/YodaController.groovy:797]: Fixed in b44cda1

File: app/grails-app/controllers/com/k_int/kbplus/OrganisationController.groovy (hunk 4 above)
Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/OrganisationController.groovy:705]: Implicitely given by propertyDefinition

**Pull request merged**